### PR TITLE
Update Splitwise to use OAuth

### PR DESCRIPTION
**Pull request by danpalmer**
## Description

Ports Splitwise to use OAuth removing the need for a somewhat complex setup process.

~Could someone from Raycast run a new instance of the PKCE proxy?~ I've now done this too. This PR works and is ready to go.

Additionally:
- Made default currency configurable.
- Added Australian dollar support.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by danpalmer** [comment]
@DiogoFerrer would you mind taking a look at this? I think it's a meaningful improvement in the usability and might help grow the userbase a bit.

File: extensions/Splitwise/src/add-expense.tsx [hunk 1]
```diff
@@ -143,6 +150,7 @@ function FillForm(props: FriendOrGroupProps) {
         <Form.Dropdown.Item value="EUR" title={`EUR (${getCurrency_code("EUR")})`} icon="💶" />
         <Form.Dropdown.Item value="GBP" title={`GBP (${getCurrency_code("GBP")})`} icon="💷" />
         <Form.Dropdown.Item value="JPY" title={`JPY (${getCurrency_code("JPY")})`} icon="💴" />
+        <Form.Dropdown.Item value="AUD" title={`JPY (${getCurrency_code("AUD")})`} icon="💵" />
```
Comment (maximilianzuern) [inline extensions/Splitwise/src/add-expense.tsx:153]: Small typo.
Should be `AUD` for Australian Dollars.

Comment (maximilianzuern) [inline extensions/Splitwise/src/add-expense.tsx:153]:
```suggestion
        <Form.Dropdown.Item value="AUD" title={`AUD (${getCurrency_code("AUD")})`} icon="💵" />
```

**Review by maximilianzuern — CHANGES_REQUESTED** [review]
I tested the PR locally. It works like a charm! 
There is just a little typo in the add-expense.tsx file.

Since I don't have experience with 0Auth, where does the `https://raycast-pkce-proxy-splitwise.fly.dev`come from? Is it hosted by Raycast?

File: extensions/Splitwise/src/add-expense.tsx (hunk 1 above)
Comment (danpalmer) [reply extensions/Splitwise/src/add-expense.tsx]: Whoops thanks!

**Comment by danpalmer** [comment]
Thanks for the review!

> Since I don't have experience with 0Auth, where does the https://raycast-pkce-proxy-splitwise.fly.devcome from? Is it hosted by Raycast?

This is currently hosted by me – it's an instance of https://github.com/mathieudutour/pkce-proxy which is from one of the Raycast team. Raycast do host a number of these, but it takes some back and forth to get them running, and in this case I had to do a bunch of server-side debugging to get the client-side working so I'm glad I hosted my own.

I'd be happy to switch over to a Raycast hosted one, but so far this process has been ad-hoc. It's at a cost to them in terms of time and money, so I'm not sure how they decide whether this is worth it or not. I imagine this would take a while to organise and get set up, so I'd prefer to stick with this for now.

Security-wise, this protects against bad actors being able to authenticate with the Splitwise API as this extension and potentially get the extension banned. It does require trust in the proxy, but no more than the trust in the extension to manage any tokens correctly.

**Review by maximilianzuern — APPROVED** [review]
LGTM!

**Comment by thomaslombart** [comment]
Hey @danpalmer, we usually host PKCE proxies to make sure no one is tampering with the authentication flow. It's true it involves some back and forth but we're here to help anyway 😊

Could you send me a message in the Slack community with the details so that I can deploy a proxy for you?

**Comment by danpalmer** [comment]
Ah! Sorry, this got lost. I can definitely pick this up again.

**Comment by maximilianzuern** [comment]
> Ah! Sorry, this got lost. I can definitely pick this up again.

Awesome! 🎉 
Just keep in mind that I made a PR #10613 in the meantime. It fetches the default currency via the API and adds it to the dropdown menu, so manual selection via preference values is no longer necessary.

**Pull request closed without merging**